lib/kubernetes-deploy/logger.rb

```diff
@@ -3,43 +3,43 @@
 
 module KubernetesDeploy
   module Logger
-    def self.included(base)
-      base.extend(ClassMethods)
-    end
+    def self.build(namespace, context, stream = $stderr, verbose_tags: false)
+      l = ::Logger.new(stream)
+      l.level = level_from_env
 
-    module ClassMethods
-      def logger=(other)
-        @logger = other
-      end
+      l.formatter = proc do |severity, datetime, _progname, msg|
+        middle = verbose_tags ? "[#{context}][#{namespace}]" : ""
+        colorized_line = ColorizedString.new("[#{severity}][#{datetime}]#{middle}\t#{msg}\n")
 
-      def logger
-        @logger ||= begin
-          l = ::Logger.new($stderr)
-          l.level = level_from_env
-          l.formatter = proc do |severity, datetime, _progname, msg|
-            log_text = "[#{severity}][#{datetime}]\t#{msg}"
-            case severity
-            when "FATAL" then "\033[0;31m#{log_text}\x1b[0m\n" # red
-            when "ERROR", "WARN" then "\033[0;33m#{log_text}\x1b[0m\n" # yellow
-            when "INFO" then "\033[0;36m#{log_text}\x1b[0m\n" # blue
-            else "#{log_text}\n"
-            end
-          end
-          l
+        case severity
+        when "FATAL"
+          colorized_line.red
+        when "ERROR", "WARN"
+          colorized_line.yellow
+        when "INFO"
+          msg =~ /^\[(KUBESTATUS|Pod)/ ? colorized_line : colorized_line.blue
```

Comment:
This matching is a temporary hack to keep from over-coloring the output if this is deployed without the output revamp PR, which will remove the hack.